frontend/packages/dev-console/src/components/edit-application/EditApplicationForm.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues } from 'formik';
+import { ModalTitle, ModalBody, ModalSubmitFooter } from '@console/internal/components/factory';
+import { BuildStrategyType } from '@console/internal/components/build';
+import GitSection from '../import/git/GitSection';
+import BuilderSection from '../import/builder/BuilderSection';
+import DockerSection from '../import/git/DockerSection';
+import AdvancedSection from '../import/advanced/AdvancedSection';
+import AppSection from '../import/app/AppSection';
+import { EditApplicationFormProps } from './edit-application-types';
+
+const EditApplicationForm: React.FC<FormikProps<FormikValues> & EditApplicationFormProps> = ({
+  handleSubmit,
+  handleReset,
+  values,
+  builderImages,
+  dirty,
+  errors,
+  status,
+  isSubmitting,
+}) => (
+  <form className="modal-content" onSubmit={handleSubmit}>
+    <ModalTitle>Edit Application</ModalTitle>
+    <ModalBody>
+      {!_.isEmpty(values.build.strategy) && <GitSection />}
```

Comment:
Is it better to check for Source strategy here?

Reply:
We show `GitSection` for both strategies `Source` and `Docker` and hence I am just checking if there is a strategy show `GitSection` and if empty do not show it.